splunk-otel-android/src/main/java/com/splunk/rum/NamedTrackableTracer.java

```diff
@@ -119,13 +135,17 @@ public void endSpanForActivityResumed() {
 
     @Override
     public void endActiveSpan() {
+        Span startupSpan = SplunkRum.getStartupTimer().getStartupSpan();
+        if (startupSpan != null) {
+            startupSpan.end();
+        }
```

Comment:
Why not simply call `SplunkRum.getStartupTimer().end()`? Is there any reason behind duplicating a part of that method?

Reply:
Also, would it make sense to inject the `AppStartupTimer` via the constructor instead of using the global?

Reply:
> Why not simply call `SplunkRum.getStartupTimer().end()`? Is there any reason behind duplicating a part of that method?

heh. I actually had this initially, and thought it was even more "blindly doing the dumb thing", and add the explicit check. Honestly, I think both are bad options. ;)

Reply:
ok, changed to injecting the startup timer. Next step (next PR) will to break the NamedTrackableTracer into 2 classes, one for activities, and one for fragments.